### PR TITLE
[WIP][TASK] Convert Once suite of ViewHelpers to static callable

### PR DESCRIPTION
**Pull request by NamelessCoder**

### CODE REVIEW

File: Classes/ViewHelpers/Once/AbstractOnceViewHelper.php [hunk 1]
```diff
@@ -52,88 +55,63 @@ public function initializeArguments()
...
+     * @param array|null $arguments
+     * @return boolean
      */
-    protected function getIdentifier()
+    protected static function evaluateCondition($arguments = null)
     {
-        if (true === isset($this->arguments['identifier'])) {
-            return $this->arguments['identifier'];
+        $identifier = static::getIdentifier($arguments);
+        $verdict = !static::assertShouldSkip($identifier);
```
Comment (bjo3rnf) [inline Classes/ViewHelpers/Once/AbstractOnceViewHelper.php]: Nice variable naming.

File: Classes/ViewHelpers/Once/AbstractOnceViewHelper.php [hunk 2]
```diff
@@ -52,88 +55,63 @@ public function initializeArguments()
...
-     * @return string
+     * @param array|null $arguments
+     * @return boolean
      */
-    protected function getIdentifier()
+    protected static function evaluateCondition($arguments = null)
     {
-        if (true === isset($this->arguments['identifier'])) {
-            return $this->arguments['identifier'];
+        $identifier = static::getIdentifier($arguments);
```
Comment (cedricziel) [inline Classes/ViewHelpers/Once/AbstractOnceViewHelper.php]: Missing argument `$renderingContext`

File: Classes/ViewHelpers/Once/AbstractOnceViewHelper.php [hunk 3]
```diff
@@ -52,88 +55,63 @@ public function initializeArguments()
...
+        $verdict = !static::assertShouldSkip($identifier);
+        if ($verdict) {
+            $GLOBALS['TSFE']->no_cache = 1;
+            static::removeIfExpired($identifier, $arguments['ttl']);
+        } else {
+            static::storeIdentifier($identifier);
```
Comment (cedricziel) [inline Classes/ViewHelpers/Once/AbstractOnceViewHelper.php]: The call to storeIdentifier() misses a required argument $arguments.

**Comment by NamelessCoder** [comment]
Looks like I deleted my fork a bit too early :/ can be restored in any fork if you need to continue it before I get around to it!

**Pull request closed without merging**